src/conmon.c

```diff
@@ -464,24 +512,14 @@ int main(int argc, char *argv[])
 	 * the container runs.  Close them before we notify the container exited, so that they can be
 	 * reused immediately.
 	 */
-	DIR *fdsdir = opendir("/proc/self/fd");
-	if (fdsdir != NULL) {
-		int fd;
-		int dfd = dirfd(fdsdir);
-		struct dirent *next;
-
-		for (next = readdir(fdsdir); next; next = readdir(fdsdir)) {
-			const char *name = next->d_name;
-			if (name[0] == '.')
-				continue;
-
-			fd = strtoll(name, NULL, 10);
-			if (fd == dfd || fd == sync_pipe_fd || fd == attach_pipe_fd || fd == dev_null_r || fd == dev_null_w)
-				continue;
-			close(fd);
-		}
-		closedir(fdsdir);
-	}
+	int fd;
+
+	for (fd = 3; fd < open_files_max_fd; fd++)
+		if (open_files_set == NULL || FD_ISSET (fd % FD_SETSIZE, &(open_files_set[fd / FD_SETSIZE])))
+			if (fd == sync_pipe_fd || fd == attach_pipe_fd || fd == dev_null_r || fd == dev_null_w
+			    || fd == oom_cgroup_fd || fd == oom_event_fd)
+				close (fd);
+	close_all_readers();
```

Comment:
and this could all also live in that new file

Reply:
same